salesforce_api/kafka_salesforce_integration/kafka-salesforce-pricebook_update/sample.bal

```diff
@@ -0,0 +1,44 @@
+import ballerinax/kafka;
+import ballerinax/salesforce as sfdc;
+
+configurable string salesforceAccessToken = ?;
+configurable string salesforceBaseUrl = ?;
+public type ProductPrice readonly & record {|
+    string name;
+    float unitPrice;
+|};
+
+public type ProductPriceUpdate readonly & record {|
+    float UnitPrice;
+|};
+
+listener kafka:Listener orderListener = new (kafka:DEFAULT_URL, {
+    groupId: "order-group-id",
+    topics: "foobar"
+});
+
+const string PRICEBOOKID = "";
+
+final sfdc:Client sfdcClient = check new ({
+    baseUrl: salesforceBaseUrl,
+    auth: {
+        token: salesforceAccessToken
+    }
+});
+
+service on orderListener {
+    isolated remote function onConsumerRecord(ProductPrice[] prices) returns error? {
+        foreach ProductPrice {name, unitPrice} in prices {
+            stream<record{},error?> retrievedStream = check sfdcClient->query(
```

Comment:
Isn't it more clear if we define a record type than using `record{}`

Reply:
For the salesforce connector when the query function is used as dependantly typed, it gives the error ```error: java.lang.ClassCastException {"message":"class io.ballerina.runtime.internal.types.BTypeReferenceType cannot be cast to class io.ballerina.runtime.api.types.RecordType (io.ballerina.runtime.internal.types.BTypeReferenceType and io.ballerina.runtime.api.types.RecordType are in unnamed module of loader 'app')"}
        at ballerinax.salesforce.7.Client:streamConverter(client.bal:358)
           ballerinax.salesforce.7.Client:processGetQueryResult(client.bal:330)```. This issue is tracked in [issue](https://github.com/ballerina-platform/ballerina-extended-library/issues/559)